libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -505,7 +512,14 @@ method publish*(g: GossipSub,
         Message.init(none(PeerInfo), data, topic, none(uint64), false)
       else:
         Message.init(some(g.peerInfo), data, topic, some(g.msgSeqno), g.sign)
-    msgId = g.msgIdProvider(msg)
+    msgIdResult = g.msgIdProvider(msg)
+
+  if msgIdResult.isErr:
+    debug "Dropping message due to failed message id generation",
```

Comment:
```suggestion
    debug "Error generating message id, skipping publish",
```

Reply:
And theses ones should probably be some INFOs, actually

Reply:
no, that would open up an attack vector: generating a message id happens _before_ pretty much every other validation like deduplication etc - in fact, it's where an untrusted external blob first meets the application and we need to treat it with care - in eth2, the message id generation function is quite complex and if it fails, it's usually due to an attack - therefore, rather than a string, I see a ValidationResult (or similar) `enum` as the way forward so that libp2p can penalise the peer for sending it.

Reply:
That's in publish, so to hit this one, it's the application doing something bad

Though, good point for the ones in rpcHandler. Maybe switch to a trace and add a metric instead

Reply:
right, this is publish - hm, yeah, if it fails here, it's almost the case that it warrants an error being passed on to the caller - I'm not sure callers have an adequate error handling policy at this stage though because `publish` is sort of best-effort..

this reminds me of https://github.com/status-im/nimbus-eth2/issues/3269 which basically says that libp2p is the wrong abstraction level to be making log level decisions

Reply:
up on this

Reply:
@Menduist changed to `trace` and metric added 👍